cranelift/codegen/src/isa/s390x/lower/isle.rs

```diff
@@ -126,7 +126,7 @@ impl generated_code::Context for IsleContext<'_, '_, MInst, Flags, IsaFlags, 6>
     fn defs_init(&mut self, abi: &Sig) -> CallRetList {
         // Allocate writable registers for all retval regs, except for StructRet args.
         let mut defs = smallvec![];
-        for i in 0..self.lower_ctx.sigs()[*abi].num_rets() {
+        for i in 0..self.lower_ctx.sigs().num_rets(*abi) {
```

Comment:
Pre-existing, but since `Sig` is now a newtype wrapper around `u32` and is `Copy`, we should probably just take an arg `abi: Sig` and avoid this dereference.

Reply:
I had @tnachen make some changes along these lines in a previous PR, but I decided not to ask for that in function signatures that weren't otherwise changed in this PR.

It would be a good follow-up PR to fix all the functions that currently take `&Sig` to take `Sig` instead. I currently see examples in `cranelift/codegen/src/isa/s390x/lower/isle.rs`, `cranelift/codegen/src/machinst/abi.rs`, and `cranelift/codegen/src/machinst/isle.rs`.

Reply:
Sg will make these changes after this one merged

Reply:
I try to make this change but looks like &Sig is generated from isle as generated code, and all non-primitive types declared as arg funcs in isle seems to be all generated as references. Should we leave this or try to change how isle code gen works?

Reply:
Or change (type Sig extern (enum)) into (type Sig (primitive u32)) ?

Reply:
Actually that doesn't work as it only generate args with u32 and not Sig anymore

Reply:
Oh, I see! I just learned how ISLE decides when to borrow last week while I've been rewriting its code generation, so this hadn't occurred to me.

It isn't important to get rid of the `&Sig` borrows. But if you'd like to try working on the ISLE compiler, I think I can now guide you through fixing this. Let me know if you're interested; if so, I'll write up a plan.

Reply:
@jameysharp I'm game! I think I kind of have a idea looking at a codebase, but I'm still new :)

Reply:
So I looked into this more, and I think this doesn't need ISLE compiler changes. I think it should work to change `(type Sig extern (enum))` to `(type Sig (primitive Sig))` and then fix the hand-written parts to match. Do you want to give that a try?